connector/src/main/scala/com/microsoft/kusto/spark/datasource/KustoFilter.scala

```diff
@@ -7,18 +7,18 @@ import org.apache.spark.sql.types._
 
 private[kusto] object KustoFilter {
   // Augment the original query to include column pruning and filtering
-  def pruneAndFilter(originalSchema: StructType, originalQuery: String, filtering: KustoFiltering): String = {
-    originalQuery + KustoFilter.buildFiltersClause(originalSchema, filtering.filters) + KustoFilter.buildColumnsClause(filtering.columns)
+  def pruneAndFilter(kustoSchema: KustoSchema, originalQuery: String, filtering: KustoFiltering): String = {
+    originalQuery + KustoFilter.buildFiltersClause(kustoSchema.sparkSchema, filtering.filters) + KustoFilter.buildColumnsClause(filtering.columns, kustoSchema.timespanColumns)
   }
 
   def pruneSchema(schema: StructType, columns: Array[String]): StructType = {
     val fieldMap = schema.fields.map(x => x.name -> x).toMap
     new StructType(columns.map(name => fieldMap(name)))
   }
 
-  def buildColumnsClause(columns: Array[String]): String = {
-    if(columns.isEmpty) "" else {
-      " | project " + columns.map(col=>s"['$col']").mkString(", ")
+  def buildColumnsClause(columns: Array[String], timespanColumns: Set[String]): String = {
```

Comment:
name here should be more aligned with what is happening, something like `toStringColumns`.

Reply:
called it toStringCastedColumns